src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs

```diff
@@ -537,11 +539,7 @@ public static int BinarySearch<T>(this ImmutableArray<T> array, int index, int l
         /// <param name="items">The array to use or copy from. May be null for "default" arrays.</param>
```

Comment:
Yup, will fix.  The only caller of this internal method is already doing the check; no need to do it twice.